src/language-js/needs-parens.js

```diff
@@ -225,6 +225,24 @@ function needsParens(path, options) {
     return true;
   }
 
+  if (
+    (node.type === "BinaryExpression" ||
+      node.type === "LogicalExpression" ||
+      node.type === "ArrowFunctionExpression" ||
+      node.type === "AwaitExpression" ||
+      node.type === "YieldExpression" ||
+      node.type === "SequenceExpression" ||
+      node.type === "ConditionalExpression" ||
+      node.type === "AssignmentExpression" ||
+      node.type === "TSAsExpression" ||
+      node.type === "TSTypeAssertion") &&
+    parent.type === "UnaryExpression" &&
+    node.comments &&
+    node.comments.length > 0
+  ) {
+    return false;
+  }
```

Comment:
I don't think it's the right approach. The `needsParens` function starts to lie here which makes it less reusable.